clients/js/src/utilities.ts

```diff
@@ -76,55 +70,57 @@ export async function createTokenAccountTx({
     owner,
   });
 
-  // Build and send the transaction
-  const tx = pipe(
-    createTransactionMessage({ version: 0 }),
-    tx => setTransactionMessageFeePayerSigner(payer, tx),
-    tx => setTransactionMessageLifetimeUsingBlockhash(blockhash, tx),
-    tx => appendTransactionMessageInstructions([createAccountIx, initializeAccountIx], tx),
-  );
-
   return {
-    tx,
+    ixs: [createAccountIx, initializeAccountIx],
     keyPair,
   };
 }
 
-export interface CreateEscrowAccountTxArgs {
+export interface CreateEscrowAccountArgs {
   rpc: Rpc<GetAccountInfoApi & GetMinimumBalanceForRentExemptionApi>;
-  blockhash: {
-    blockhash: Blockhash;
-    lastValidBlockHeight: bigint;
-  };
   payer: KeyPairSigner;
   unwrappedMint: Address;
   wrappedTokenProgram: Address;
 }
 
-export interface CreateEscrowAccountTxResult {
-  tx: CompilableTransactionMessage & TransactionMessageWithBlockhashLifetime;
-  keyPair: KeyPairSigner;
-}
+export type CreateEscrowAccountResult =
+  | {
+      address: Address;
+      ixs: IInstruction[];
+      exists: boolean;
+    }
+  | Account<Token>;
```

Comment:
I think it probably makes sense to remove the `exists` field here, if I'm understanding it correctly. If when you fetch the account it doesn't exist already, then you return the instructions to create it, with `exists: false`. If it does then you just return the account. So I don't think there's any case where `exists` will be true, in the first variant.

I think it might be clearer to do something like:

```ts
| {
  kind: 'already_exists';
  account: Account<Token>;
}
| {
  kind: 'instructions_to_create';
  address: Address;
  ixs: IInstruction[];
}
```

The `kind` isn't strictly necessary, you could just check `if ('ixs' in result)` for example to narrow the type, but it's a pattern we use elsewhere and tends to make it a bit clearer IMO.

Reply:
I find that API to be quite clear! Will update that 👍